packages/lesswrong/components/forumEvents/ForumEventPoll.tsx

```diff
@@ -300,102 +414,120 @@ const PollQuestion = ({event, classes}: {
  * If a postId is provided, we just give points to that post (ex. when on a post page).
  * Otherwise, we open a modal.
  */
-export const ForumEventPoll = ({postId, hideViewResults, classes}: {
-  postId?: string,
-  hideViewResults?: boolean,
-  classes: ClassesType<typeof styles>,
+export const ForumEventPoll = ({
+  postId,
+  hideViewResults,
+  classes,
+}: {
+  postId?: string;
+  hideViewResults?: boolean;
+  classes: ClassesType<typeof styles>;
 }) => {
-  const {currentForumEvent: event, refetch} = useCurrentForumEvent()
-  const {onSignup} = useLoginPopoverContext()
-  const currentUser = useCurrentUser()
-
-  const initialUserVotePos: number|null = getForumEventVoteForUser(event, currentUser)
-  // The actual x position of the left side of the user's vote circle,
-  // as a number between 0 and 0.976 (i.e. (SLIDER_MAX_WIDTH - USER_IMAGE_SIZE) / SLIDER_MAX_WIDTH)
-  const [votePos, setVotePos] = useState<number>(initialUserVotePos ?? defaultVotePos)
-  // The x position of the vote in the db
-  const [currentUserVote, setCurrentUserVote] = useState<number|null>(initialUserVotePos)
-  const hasVoted = currentUserVote !== null
-  
-  const sliderRef = useRef<HTMLDivElement|null>(null)
-  // Whether or not the user is currently dragging their vote
-  const [isDragging, setIsDragging] = useState(false)
-
-  // Whether or not the poll results (i.e. other users' votes) are visible.
-  // They are hidden until the user clicks on "view results".
-  const [resultsVisible, setResultsVisible] = useState(false)
-  const [voteCount, setVoteCount] = useState(event?.voteCount ?? 0)
-  
-  // Get profile image and display name for all other users who voted, to display on the slider
+  const { currentForumEvent: event, refetch } = useCurrentForumEvent();
+  const { onSignup } = useLoginPopoverContext();
+  const currentUser = useCurrentUser();
+
+  const initialUserVotePos: number | null = getForumEventVoteForUser(
+    event,
+    currentUser
+  );
+  const initialBucketIndex =
+    initialUserVotePos !== null
+      ? Math.round(initialUserVotePos * (NUM_TICKS - 1))
+      : DEFAULT_VOTE_INDEX;
+  const [currentBucketIndex, setCurrentBucketIndex] = useState<number>(
+    initialBucketIndex
+  );
+  const [currentUserVote, setCurrentUserVote] = useState<number | null>(
+    initialUserVotePos
+  );
+  const hasVoted = currentUserVote !== null;
+  const isDragging = useRef(false);
+
+  const sliderRef = useRef<HTMLDivElement | null>(null);
+  const tickRefs = useRef<(HTMLDivElement | null)[]>([]);
+
+  const [resultsVisible, setResultsVisible] = useState(false);
+  const [voteCount, setVoteCount] = useState(event?.voteCount ?? 0);
+
+  const votersRef = useRef<UserOnboardingAuthor[]>([])
```

Comment:
Oh sorry I meant to add a comment here. It's because `voters` is briefly undefined when refetching which causes the graph to flicker. I don't actually understand what causes that (I thought it kept the previous result until the new one came through), but decided not to look too much into it and just do this